constraints.txt

```diff
@@ -0,0 +1,2 @@
+git+git://github.com/david4096/ga4gh-client.git@protobuf31#egg=ga4gh_client
```

Comment:
Yeah, good catch both!

I like the idea of keeping the constraints file out of the tree and it being part of the development documentation. The problem I ran into was getting the dependencies to resolve against pinned versions installed as part of `install_requires`. You have to create a properly formed [egg fragment](http://stackoverflow.com/questions/11835396/why-egg-foo-when-pip-installing-from-git-repo) at the end of the url, which was a task I didn't really want to hand down to other developers.

The alternative was to unpin the version in requirements and force a dependency against just the name in the fragment you see. That suggests that I was headed towards @andrewjesaitis' solution, which would have the master branch `constraints.txt`, when possible, against a released version of the schemas. The entries in constraints have the same syntax as in the `requirements.txt`. 

We don't do a release every time something is merged into the schemas, so the `constraints.txt` should probably be against schemas/master until a release. As long as PRs that have dependencies on other repositories are merged within a similar time, a release coordinator will easily be able to pick versions that are stable across our dependencies.

To do this, first the release coordinator makes a schema release with a tagged version. They then modify the `requirements.txt` of the client to point at the newly released version of schemas and remove the contents of the `constraints.txt`. They then release the client and repeat the same process in the server. I think this is more or less the same onerousness of the task that assembling the egg fragment would be, but done only once per release by the release coordinator, and could be automated.

I am trying to optimize the development experience of modifying the schemas and sharing an implementation. This requires nailing down the release process for the PyPi package. If we did keep the constraints file empty in master, wouldn't we need to release the package to pip with every schema change that needed implementation?